packit/base_git.py

```diff
@@ -359,14 +359,13 @@ def set_specfile_content(self, specfile: Specfile, version: str, comment: str):
         :param version: version to set in self.specfile
         :param comment: new comment for the version in %changelog
         """
-        this_changelog = self.specfile.spec_content.section("%changelog")
-        this_version = self.specfile.get_version()
+        provided_changelog = specfile.spec_content.section("%changelog")
         self.specfile.spec_content.sections[:] = specfile.spec_content.sections[:]
         self.specfile.save()
-        self.specfile.spec_content.replace_section("%changelog", this_changelog)
-        self.specfile.set_version(this_version)
+        self.specfile.spec_content.replace_section("%changelog", provided_changelog)
         self.specfile.save()
-        self.specfile.set_spec_version(version=version, changelog_entry=comment)
+        if version and comment:
+            self.specfile.set_spec_version(version=version, changelog_entry=comment)
```

Comment:
If we expect  `version` to be optional, let's update the header and annotation.

Do I get it correctly, that if we want to change/update the version, we must add a new changelog entry? If so, I am ok with this approach. Otherwise, we can add the following two lines:

```python
if version and not comment:
    self.specfile.set_version(version)
```

Reply:
sorry, my bad, now I checked the underlying function and it already checks if version and changelog_entry are set, so the if is indeed redundant